### PR TITLE
fix(aws_s3 source): Handle s3:TestEvent and drop it

### PR DESCRIPTION
**Pull request by bencord0**
https://github.com/vectordotdev/vector/discussions/14047

Vector currently fails to process test events created by S3. These are benign events that can be ignored to prevent serde parser failures.

### CODE REVIEW

File: src/sources/aws_s3/sqs.rs [hunk 1]
```diff
@@ -379,12 +379,20 @@ impl IngestorProcess {
...
+        let s3_event: Event = serde_json::from_str(message.body.unwrap_or_default().as_ref())
             .context(InvalidSqsMessageSnafu {
-                message_id: message.message_id.unwrap_or_else(|| "<empty>".to_owned()),
+                message_id: message.message_id.clone().unwrap_or_else(|| "<empty>".to_owned()),
             })?;
 
-        self.handle_s3_event(s3_event).await
+        match s3_event {
+            Event::TestEvent(_s3_test_event) => {
+                debug!(message = "Found S3 Test Event", ?message.message_id);
```
Comment (spencergilbert) [inline src/sources/aws_s3/sqs.rs:389]:
```suggestion
                debug!(?message.message_id, message = "Found S3 Test Event.");
```

Passes the existing `check-events` scripts, and re-ordered to pass an upcoming change to the script.

File: src/sources/aws_s3/sqs.rs [hunk 2]
```diff
@@ -592,6 +600,21 @@ impl IngestorProcess {
     }
 }
 
+#[derive(Clone, Debug, Deserialize)]
+enum Event {
+    TestEvent(S3TestEvent),
+    Event(S3Event),
+}
+
+// https://docs.aws.amazon.com/AmazonS3/latest/userguide/how-to-enable-disable-notification-intro.html
```
Comment (spencergilbert) [inline src/sources/aws_s3/sqs.rs:609]: I'm not entirely sure how this link relates to the test event - it seems like the `notification-content-structure.html` below also covers this event. Perhaps that comment should just get pulled up to the new `Enum`?

**Review by spencergilbert — CHANGES_REQUESTED** [review]
Thanks for the contribution, looks like a straightforward fix for an annoying issue.

**Comment by bencord0** [comment]
Thanks for the review @spencergilbert, I've updated the patch with both of those changes

**Review by spencergilbert — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
Looks good, thanks for this @bencord0 !

**Comment by bencord0** [comment]
I've pushed a change based on the failed [`make check-fmt`](https://github.com/vectordotdev/vector/actions/runs/3135816207/jobs/5094336163) that adjusts the formatting.

**Comment by spencergilbert** [comment]
🤔 looks like there's still a `cargo fmt` issue?

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**